api/src/main/scala/za/co/absa/hyperdrive/ingestor/api/context/HyperdriveContext.scala

```diff
@@ -15,24 +15,23 @@
 
 package za.co.absa.hyperdrive.ingestor.api.context
 
-import scala.collection.{mutable => mu}
 import scala.reflect.ClassTag
-import scala.util.{Failure, Success, Try}
 import scala.reflect.runtime.{universe => ru}
+import scala.util.Try
 
 object HyperdriveContext {
-  private val contextValues = mu.Map[String, Any]()
+  private var contextValues = Map[String, Any]()
 
-  def put(key: String, value: Any): Option[Any] = contextValues.put(key, value)
-
-  def get[T:ClassTag:ru.TypeTag](key: String): Try[T] = {
-    val value: Try[Any] = contextValues.get(key)
-      .map(Success(_))
-      .getOrElse(Failure(new NoSuchElementException(s"Could not find value for key $key")))
+  def put(key: String, value: Any): Unit = {
+    contextValues = contextValues + (key -> value)
+  }
 
-    value.flatMap {
-      case asT: T => Success(asT)
-      case _ => Failure(new ClassCastException(s"Could not cast value $value for key $key to type ${ru.typeOf[T]}"))
-    }
+  def get[T: ClassTag : ru.TypeTag](key: String): Try[T] = {
+    Try(
```

Comment:
This is a lovely piece of code 👍

Reply:
Hm, just out of curiosity, how does it behave, when the values is of type `Int` but requested `Long`?

Reply:
Thanks :). If you request an `Int` as a `Long` it throws the ClassCastException since you can't cast `Int` to `Long` in Scala with a pattern match.

Reply:
> This is a lovely piece of code 👍

I agree !